src/elements/content-sidebar/activity-feed/annotation-thread/useRepliesAPI.js

```diff
@@ -0,0 +1,128 @@
+// @flow
+import uniqueId from 'lodash/uniqueId';
+import APIFactory from '../../../../api';
+import commonMessages from '../../../common/messages';
+import { commentsErrors } from './errors';
+
+import type { BoxItemPermission, User } from '../../../../common/types/core';
+import type { BoxCommentPermission, Comment, FeedItemStatus } from '../../../../common/types/feed';
+import type { ElementsXhrError } from '../../../../common/types/api';
+
+type Props = {
+    annotationId: string,
+    api: APIFactory,
+    currentUser: User,
+    fileId: string,
+    filePermissions: BoxItemPermission,
+    handleAddReplyItem: Comment => void,
+    handleRemoveReplyItem: (id: string) => void,
+    handleUpdateReplyItem: (updatedValues: Object, id: string) => void,
+};
+
+const useRepliesAPI = ({
+    annotationId,
+    api,
+    currentUser,
+    fileId,
+    filePermissions,
+    handleAddReplyItem,
+    handleRemoveReplyItem,
+    handleUpdateReplyItem,
+}: Props) => {
+    const setReplyPendingStatus = (replyId: string, pendingStatus: boolean) => {
+        handleUpdateReplyItem({ isPending: pendingStatus }, replyId);
+    };
+
+    const addNewPendingReply = (baseReply: Object) => {
+        const date = new Date().toISOString();
+        const pendingReply: Comment = {
+            created_at: date,
+            created_by: currentUser,
+            modified_at: date,
+            isPending: true,
+            ...baseReply,
+        };
+
+        handleAddReplyItem(pendingReply);
+    };
+
+    const createReplySuccessCallback = (replyId: string, reply: Comment) => {
+        handleUpdateReplyItem(
+            {
+                ...reply,
```

Comment:
Minor, but it looks like this is already handled by `handleUpdateReplyItem`.

Reply:
I've rather to keep it to always update comment with recently fetched data